sa/model.go

```diff
@@ -59,6 +61,51 @@ func badJSONError(msg string, jsonData []byte, err error) error {
 
 const regFields = "id, jwk, jwk_sha256, contact, agreement, initialIP, createdAt, LockCol, status"
 
+// ClearEmail removes the provided email address from one specified registration. If
+// there are multiple email addresses present, it does not modify other ones. If the email
+// address is not present, it does not modify the registration and return nil error.
+func ClearEmail(dbMap db.DatabaseMap, ctx context.Context, regID int64, email string) error {
+	_, overallError := db.WithTransaction(ctx, dbMap, func(txWithCtx db.Executor) (interface{}, error) {
+		curr, err := selectRegistration(txWithCtx, "id", regID)
+		if err != nil {
+			return nil, err
+		}
+
+		currPb, err := registrationModelToPb(curr)
+		if err != nil {
+			return nil, err
+		}
+
+		var newContacts []string
+		modified := false
+		for _, contact := range currPb.Contact {
+			if contact == "mailto:"+email {
+				fmt.Fprintf(os.Stderr, "Skipping %s\n", contact)
+				modified = true
```

Comment:
I think `modified = true` should be in the else statement instead because if it's being skipped, that implies no work is to be done on that record. This would need a corresponding change to the integration test though.
```
$ ./t.sh -i -f TestAdminClearEmail
--- FAIL: TestAdminClearEmail (0.05s)
    admin_test.go:39: clear-email output: 16:12:20.484387 6 admin 1-Hk4QM [AUDIT] Searching database for accounts with email addresses matching "example@mail.example.letsencrypt.org" in order to clear the email addresses
        Skipping mailto:example@mail.example.letsencrypt.org
        Skipping mailto:example@mail.example.letsencrypt.org
        
    admin_test.go:44: [mailto:long-example@mail.example.letsencrypt.org mailto:third-example@mail.example.letsencrypt.org]
    admin_test.go:50: [[mailto:example@mail.example.letsencrypt.org]] !(deep)= [[]]
FAIL
FAIL    github.com/letsencrypt/boulder/test/integration 0.087s
FAIL
```

Reply:
The logic here was a bit confusing (I actually wrote it backwards the first time too). I modified it to be clearer.